test/test_integrate.py

```diff
@@ -545,3 +545,12 @@ def f(args):
     val_eps = f(1.0 + eps)
     numerical_grads = (val_eps - val) / eps
     assert shaped_allclose(grads, numerical_grads)
+
+
+def test_complex_dtypes():
+    # Test that we can solve complex ODEs
+    term = diffrax.ODETerm(lambda t, y, args: -1.0j * y)
+    solver = diffrax.Tsit5()
+    y0 = jnp.array([1.0, 1.0j])
+    sol = diffrax.diffeqsolve(term, solver, t0=0, t1=1, dt0=0.1, y0=y0)
+    assert jnp.allclose(sol.ys[0], jnp.exp(-1.0j) * y0)
```

Comment:
Thanks for your comments!
I've added tests for different solvers, and I noticed that the implict solvers do not work. I thought to make the jacwd holomorphic if the `jnp.iscomplexobj(flat)` is evaluated as true, but it seems that it might be cast to `float32`:
https://github.com/patrick-kidger/diffrax/blob/cec091c5e4cc4311f64ae3aa09a371db5fe766ee/diffrax/nonlinear_solver/base.py#L99
(I don't understand when this branch happens)

Reply:
I think this branch is superfluous now, actually. You can probably remove it.